master/buildbot/steps/artifact.py

```diff
@@ -366,7 +423,9 @@ def start(self):
         #find artifact dependency
         triggeredbybrid = self.build.requests[0].id
         br = yield self.master.db.buildrequests.getBuildRequestTriggered(triggeredbybrid, self.artifactBuilderName)
-
+        if br is None:
```

Comment:
Make new subclass instead of this logic, perhaps `DownloadArtifactFromTriggered`?